packages/orchestration/src/transaction/transaction-orchestrator.ts

```diff
@@ -144,29 +146,90 @@ export class TransactionOrchestrator extends EventEmitter {
     }
   }
 
-  private async checkStepTimeout(transaction, step) {
+  private hasExpired(
```

Comment:
suggestion: Perhaps, this method should be documentation, even though it's private, since there are a lot of conditions going into these checks. It isn't immediately obvious when the step or transaction has actually timed out 😅